user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -290,23 +324,32 @@ private void maybeGenerateCredentials(Reconciliation reconciliation, KafkaUserMo
      * @param user              Model describing the KafkaUser
      * @param userSecret        Secret with existing user credentials or null if the secret doesn't exist yet
      */
-    private void maybeGenerateScramCredentials(Reconciliation reconciliation, KafkaUserModel user, Secret userSecret)   {
-        Secret desiredPasswordSecret = null;
+    private CompletionStage<Void> maybeGenerateScramCredentials(Reconciliation reconciliation, KafkaUserModel user, Secret userSecret) {
+        CompletableFuture<Secret> desiredPasswordPromise = new CompletableFuture<>();
 
-        if (user.isUserWithDesiredPassword())   {
+        if (user.isUserWithDesiredPassword()) {
             // User is a SCRAM-SHA-512 user and requested some specific password instead of generating a random password
-            desiredPasswordSecret = client.secrets().inNamespace(reconciliation.namespace()).withName(user.desiredPasswordSecretName()).get();
-            if (desiredPasswordSecret == null) {
-                throw new InvalidResourceException("Secret " + user.desiredPasswordSecretName() + " in namespace " + reconciliation.namespace() + " with requested password not found");
-            }
+            getRequiredSecret(
+                    reconciliation.namespace(),
+                    user.desiredPasswordSecretName(),
+                    () -> new InvalidResourceException(String.format(MISSING_SECRET_MSG, user.desiredPasswordSecretName(), reconciliation.namespace(), "requested password")))
```

Comment:
Is this really needed? The secrets are mostly self-explanatory out of the context. So I'm not sure whether this hyper-customizes message isn't just overcomplicating the code compared to a simple error message with name and namespace generated in the `getRequiredSecret` method.

Reply:
This was to allow using `getRequiredSecret` for the CA cert/key as well as the SCRAM-SHA desired password without changing the latter's exception message. If it's fine to remove the description part from all three, that would be better.

Reply:
Updated to have simpler message (secret name and namespace missing) for all potential exceptions.